### DISCUSSION TITLE
True Stock Cost when receiving purchase orders

### DISCUSSION THREAD

**Discussion by speskin97** [inventree/InvenTree#5681] [Q&A]
Hi all, 

We are just setting up our Inventree environment for our small business and I wanted to ask about how to receive Stock with more than just the purchase price included. 

For example, consider if we made a purchase where there are multiple line items, shipping, and tax included. When I'm adding line items to a Purchase Order in Inventree, I'm adding just the "Purchase price" so we can track what that cost is over time. I don't see an obvious way to also add in these shipping and tax costs where they get cooked into the final cost of the "Stock" item which gets received. So far I have been manually doing this math and separately adjusting the stock value, while leaving the PO with the original price. The math I'm doing is basically the purchase price, sales tax, and shipping based on the proportion of quantity in the order, all cooked into each individual part that gets received. This in turn helps give us a truly accurate BOM valuation. 

If I add line items to the "Extra Lines" section it doesn't seem to affect the recieved stock cost at all. 

Is there a way you guys have figured out how to set this up, or am I stuck doing the manual math for each order? 

I'm happy to share more details about the exact math with examples if anyone is curious.

**Comment by SchrodingersGat** [comment]
@speskin97 this is pretty much identical to https://github.com/inventree/InvenTree/discussions/5672

> Is there a way you guys have figured out how to set this up, or am I stuck doing the manual math for each order?

we have not figured this out yet, but we would like to. Please check out the other discussion linked above. If there is a nice way to implement this (and account for different use cases, edge conditions, etc) then I'm fully supportive.
